lib/Twig/Node/Expression/EmptyCoalesce.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+class Twig_Node_Expression_EmptyCoalesce extends \Twig_Node_Expression
+{
+
+    public function __construct(Twig_Node $left, Twig_Node $right, $lineno)
+    {
+        $left->setAttribute('ignore_strict_check', true);
+        $left->setAttribute('is_defined_test', false);
+        $right->setAttribute('ignore_strict_check', true);
+        $right->setAttribute('is_defined_test', false);
+        parent::__construct(
+            ['left' => $left, 'right' => $right],
+            ['ignore_strict_check' => true, 'is_defined_test' => false],
+            $lineno
+        );
+    }
+
+    public function compile(\Twig_Compiler $compiler)
+    {
+            $compiler
+                ->raw('((empty(')
```

Comment:
Using `empty` is a no go for me (see @stof comment about edge cases that makes `empty` far from being useful in this case).

Reply:
I'd echo the comments from @brandonkelly below; it's called the Empty Coalesce Operator, so having it behave the way `empty()` does makes sense?

Reply:
but by PHP's empty rules, or by Twig's rules ? The `is empty` test is **not** using PHP's `empty` due to all the weird edge cases it has.

Reply:
Ah interesting, I didn't know that! Is there a way to call Twig's internal `empty` from the EmptyCoalesce node parser?

Give me a pointer, and I'll get it done!

Reply:
@stof You could make that same argument about `??`, the **null**-coalescing operator, which in addition to checking if a value is null will also check if it’s set at all, despite that not being something that `var == null` would cover. And yet `??` exists in Twig. I’d say that’s enough precedence for an “**empty**-coalescing operator” to have a slightly different definition of “empty” than an unrelated `empty` test.

Reply:
Okay, I updated it to use `twig_test_empty()` and also updating the namespacing!

Hopefully this will help get it accepted?

Reply:
I wanted to call `twig_test_empty()` in `Twig\Extension\CoreExtension.php` but the method isn't exposed. So instead I just copied that function verbatim.

This obviously begs for refactoring that method out so that both the Twig `empty` test and the `???` empty coalesce operator use the same core method.

Not being that familiar with the codebase, though, I'm not sure how the maintainers would want to approach that.